### ISSUE TITLE
Missing aria-label for header checkbox in tables that use rowSelection

### ISSUE THREAD

**Issue by SpaNb4** [ant-design/ant-design#37238]
### What problem does this feature solve?

For screen reader users, it is important to understand what action will occur when clicking on a certain element. 
Ref: https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html

But if we look at this page https://ant.design/components/table/#components-table-demo-row-selection, we can see that the checkbox in the table header does not have any ```aria-label```. 
![image](https://user-images.githubusercontent.com/69504277/186629556-a5d48102-61cf-47f5-8fcf-c5526b578afe.png)

And screen reader announe it like this: "row 1  column 1     check box  not checked", that may not be clear to the user

### What does the proposed API look like?
We should set the aria-label to **"Select all"** for the header-level checkboxes

**Comment by mellis481** [comment]
@SpaNb4 https://github.com/ant-design/ant-design/pull/38336 addressed the issue you reported so you can close this.
